main_test.go

```diff
@@ -167,55 +167,306 @@ func TestApi(t *testing.T) {
 
 func TestPublishersEndpoints(t *testing.T) {
 	tests := []TestCase{
+		// GET /publishers
 		{
-			query: "GET /v1/publishers",
+			description: "GET the first page on publishers",
+			query:       "GET /v1/publishers",
+			fixtures:    []string{"publishers.yml", "publishers_code_hosting.yml"},
 
 			expectedCode:        200,
-			expectedBody:        `{"data":[]}`,
 			expectedContentType: "application/json",
+			validateFunc: func(t *testing.T, response map[string]interface{}) {
+				assert.IsType(t, []interface{}{}, response["data"])
+				data := response["data"].([]interface{})
+
+				assert.Equal(t, 25, len(data))
+
+				// Default pagination size is 25, so there's another page and
+				// next cursor should be present
+				assert.IsType(t, map[string]interface{}{}, response["links"])
+
+				links := response["links"].(map[string]interface{})
+				assert.Nil(t, links["prev"])
+				assert.Equal(t, "?page[after]=WyJmNDFmYjMzZi1hZmIxLTRhYzUtOWZmZC1kZjJmMmZhYTRmM2YiLCIyMDE4LTExLTEyVDAwOjAwOjAwWiJd", links["next"])
+
+				assert.IsType(t, map[string]interface{}{}, data[0])
+				firstPub := data[0].(map[string]interface{})
+				assert.NotEmpty(t, firstPub["email"])
+
+				assert.IsType(t, []interface{}{}, firstPub["codeHosting"])
+				assert.Greater(t, len(firstPub["codeHosting"].([]interface{})), 0)
```

Comment:
This should actually be `2`.

There's something wrong somewhere because all the `codeHosting`s of all publishers return empty lists in the API.

```suggestion
				assert.Greater(t, len(firstPub["codeHosting"].([]interface{})), 2)
```

Reply:
Uhm, no it returns data. I put 0 in the tests (i can update that value to 1), but actually the're two elements.

<img width="1290" alt="image" src="https://user-images.githubusercontent.com/5399264/187680311-422e9b52-97ee-4bbf-8a67-f74994135bd1.png">

@bfabio

Reply:
Ok, there is a bug in the POST endpoint.
Seems like the endpoint constantly update the existing codeHosting record instead of adding new ones.

Reply:
Now fixed in commit pushed in this branch.

Reply:
I read that as `assert.Equal` initially, but yeah, i'd say `assert.Equal(t, len(...), 2)`.

Gonna edit the suggestion

Reply:
Can we close that?